lib/generateNamespaceFactory.js

```diff
@@ -29,13 +30,11 @@ function generateNamespaceFactory(ns, defs) {
             .bl(`if (namespace !== '${ns.namespace}')`, () => {
               cw.ln(`throw new Error(\`Unsupported type in ${factory}: \${type}\`);`);
             })
-            .ln('switch (elementName) {');
-          for (const def of defs) {
-            const elName = className(def.identifier.name);
-            cw.ln(`case '${elName}': return ${elName}.fromJSON(json);`);
-          }
-          cw.ln(`default: throw new Error(\`Unsupported type in ${factory}: \${type}\`);`)
-            .ln('}');
+            .ln(`const klass = ClassRegistry['${ns.namespace}'][elementName];`)
+            .bl(`if (!klass)`, ()=> {
+              cw.ln(`throw new Error(\`Unsupported type in ${factory}: \${type}\`);`);
```

Comment:
It looks like `type` is not defined in the generated class anymore.  Should this be `shrType` or `elementName`?

Reply:
(I think this comment was intended for the function below) Updated to use `shrType`

Reply:
Ah yes.  My bad.  Thanks.